SampleCodeTest/TestRunner.cs

```diff
@@ -42,51 +42,66 @@ private static short GetMonth()
+                using (StreamReader reader = File.OpenText(fileName))
                 {
-                    try
+                    TestRunner tr = new TestRunner();
+                    var numRetries = 3;
+
+                    string line;
+                    while ((line = reader.ReadLine()) != null)
                     {
-                        if (isDependent.Equals("0"))
-                        {
-                            response = InvokeRunMethod(apiName);
-                        }
-                        else
+                        string[] items = line.Split('\t');
+
+                        string apiName = items[0];
+                        string isDependent = items[1];
+                        string shouldApiRun = items[2];
+
+                        if (!shouldApiRun.Equals("1"))
+                            continue;
+
+                        Console.WriteLine(new String('-', 20));
+                        Console.WriteLine("Running test case for :: " + apiName);
+                        Console.WriteLine(new String('-', 20));
+                        ANetApiResponse response = null;
+                        for (int i = 0; i < numRetries; ++i)
                         {
-                            response = (ANetApiResponse)typeof(TestRunner).GetMethod("Test" + apiName).Invoke(tr, new Object[] { });
+                            try
+                            {
+                                if (isDependent.Equals("0"))
+                                {
+                                    response = InvokeRunMethod(apiName);
+                                }
+                                else
+                                {
+                                    response = (ANetApiResponse)typeof(TestRunner).GetMethod("Test" + apiName).Invoke(tr, new Object[] { });
+                                }
+
+                                if ((response != null) && (response.messages.resultCode == messageTypeEnum.Ok))
+                                    break;
+
+                            }
+                            catch (Exception ex)
+
+                            {
+                                Console.WriteLine(ex.ToString());
+                            }
                         }
 
-                        if ((response != null) && (response.messages.resultCode == messageTypeEnum.Ok))
-                            break;
-                    }
-                    catch (Exception e)
-                    {
-                        Console.WriteLine(apiName);
-                        Console.WriteLine(e.ToString());
+                        Assert.IsNotNull(response);
+                        Assert.AreEqual(response.messages.resultCode, messageTypeEnum.Ok);
                     }
+
                 }
-                Assert.IsNotNull(response);
-                Assert.AreEqual(response.messages.resultCode, messageTypeEnum.Ok);
+
+            }
+            catch (Exception ex)
```

Comment:
Why is it necessary to change this file? I see no reason to have the flow for this disrupted like this. This implementation will break existing Travis CI flows.

Reply:
Reverted   code and pushed the changes .please review